lib/jobs/ansible-job.js

```diff
@@ -65,12 +86,29 @@ function ansibleJobFactory(
             return self._publishAnsibleResult(self.routingKey, data);
         })
         .then(function() {
+            return fs.readFileAsync(self.outFileName, 'utf8');
+        })
+        .then(function(contents) {
+            assert.arrayOfString(self.context.ansibleResultFile);
+            self.context.ansibleResultFile.push(contents);
+            fs.unlink(self.outFileName, function(err) {
```

Comment:
@stevemendes , the fs.unlink should also use the Promise, otherwise the next then will be executed before the unlink completes.

Below is the sample code:
```
.then(function() {
    return fs.readFileAsync(self.outFileName, 'utf8');
})
.then(function() {
    ...
    return fs.unlinkAsync(self.outFileName);
})
.then(function() {
    ...
    self._done();
})
...
```